src/main/java/cz/muni/fi/pv286/mmt/representation/ArrayParser.java

```diff
@@ -0,0 +1,314 @@
+                }
+                if ("".equals(extracted)) {
+                    current.setIntermediate(null);
+                    current.setValue(null);
+                    stack.add(current);
+                } else if (extracted != null) {
+                    stack.add(current);
+                    List<String> tokens = topLevelSplitTokens(extracted);
+                    for (String token : tokens) {
+                        ResultTree t = new ResultTree(token.trim());
+                        current.getChildren().add(t);
+                        stack.add(t);
+                    }
+                    current.setIntermediate(null);
+                } else {
+
+                    Matcher hexExtracted = Patterns.hexExtractor.matcher(current.getIntermediate());
+                    Matcher decimalExtracted = Patterns.decimalExtractor.matcher(current.getIntermediate());
+                    Matcher bytesExtracted = Patterns.bytesExtractor.matcher(current.getIntermediate());
+                    Matcher bitsExtracted = Patterns.bitsExtractor.matcher(current.getIntermediate());
+
+                    if (hexExtracted.find()) {
+                        extracted = hexExtracted.group(1);
+                        if (extracted.length() > 1) {
+                            byte[] v = new byte[] {HexParser.getByteFromHex(extracted.charAt(0), extracted.charAt(1))};
+                            current.setValue(v);
+                        } else {
+                            byte[] v = new byte[] {HexParser.getByteFromHex('0', extracted.charAt(0))};
+                            current.setValue(v);
+                        }
+                    } else if (decimalExtracted.find()) {
+                        extracted = decimalExtracted.group(1);
+                        int value = Integer.parseInt(extracted, 10);
+                        if (value > 255 || value < 0) {
+                            throw new InvalidArrayValueException();
+                        }
+                        byte[] v = new byte[] {(byte) value};
+                        current.setValue(v);
+                    } else if (bytesExtracted.find()) {
+                        extracted = bytesExtracted.group(1);
+                        byte[] v = new byte[] {HexParser.getByteFromHex(extracted.charAt(0), extracted.charAt(1))};
+                        current.setValue(v);
+                    } else if (bitsExtracted.find()) {
+                        extracted = bitsExtracted.group(1);
+                        byte[] v = new byte[] {Byte.parseByte(extracted, 2)};
+                        current.setValue(v);
+                    } else {
+                        throw new InvalidArrayValueException();
+                    }
+                    current.setIntermediate(null);
+                }
+
+            } else if (stack.isEmpty()) {
+                return current;
+            }
+        }
+        return null;
```

Comment:
**chore:** Refactor this method.

Too many nested if statements. This whole method is unreadable.